includes/Modules/Analytics_4/Conversion_Reporting/Conversion_Reporting_New_Badge_Events_Sync.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics_4\Conversion_Reporting\Conversion_Reporting_New_Badge_Events_Sync
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics_4\Conversion_Reporting;
+
+use Google\Site_Kit\Core\Storage\Transients;
+
+/**
+ * Class providing implementation of "new" badge for detected conversion reporting events.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Conversion_Reporting_New_Badge_Events_Sync {
+
+	/**
+	 * The detected events transient name.
+	 */
+	public const NEW_EVENTS_BADGE_TRANSIENT = 'googlesitekit_conversion_reporting_new_badge_events';
+
+	/**
+	 * Transients instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Transients
+	 */
+	protected $transients;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Transients $transients  Transients instance.
+	 */
+	public function __construct(
+		Transients $transients
+	) {
+		$this->transients = $transients;
+	}
+
+	/**
+	 * Saves new events badge to the expirable items.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $new_events New events array.
+	 */
+	public function sync_new_events_badge( $new_events ) {
+		$new_events_badge         = $this->transients->get( self::NEW_EVENTS_BADGE_TRANSIENT );
+		$save_new_badge_transient = fn( $events ) => $this->transients->set(
+			self::NEW_EVENTS_BADGE_TRANSIENT,
+			array(
+				'created_at' => time(),
+				'events'     => $events,
+			),
+			7 * DAY_IN_SECONDS
+		);
+
+		if ( ! $new_events_badge ) {
+			$save_new_badge_transient( $new_events );
+			return;
+		}
+
+		$new_events_badge_time_existed = time() - $new_events_badge['created_at'];
```

Comment:
```suggestion
		$new_events_badge_age = time() - $new_events_badge['created_at'];
```

Let's rename this to read a little cleaner, what do you think? We can also consider:

- `$new_events_badge_duration`
- `$new_events_badge_elapsed_time`